test/dynamo/test_pgo.py

```diff
@@ -0,0 +1,127 @@
+# Owner(s): ["module: dynamo"]
+
+import contextlib
+import importlib
+import os
+import sys
+
+import torch._dynamo.config
+import torch._dynamo.test_case
+import torch.compiler.config
+from torch._dynamo.testing import CompileCounter
+from torch._inductor.utils import clear_inductor_caches, fresh_inductor_cache
+
+
+# LOL.  https://github.com/pytorch/pytorch/issues/139252
+spec = importlib.util.spec_from_file_location(
+    "mock_cache", os.path.join(os.path.dirname(__file__), "../inductor/mock_cache.py")
+)
+mock_cache = importlib.util.module_from_spec(spec)
+sys.modules["mock_cache"] = mock_cache
+spec.loader.exec_module(mock_cache)
+
+
+class PgoTest(torch._dynamo.test_case.TestCase):
+    def setUp(self):
+        super().setUp()
+        self._test_stack = contextlib.ExitStack()
+        self._test_stack.enter_context(
+            torch.compiler.config.patch(workflow_id=self.id())
+        )
+        self._test_stack.enter_context(
+            torch._dynamo.config.patch(automatic_dynamic_local_pgo=True)
+        )
+        self._test_stack.enter_context(fresh_inductor_cache())
```

Comment:
any reason you dont want to extend inductor testcase here and get this for free?

also inb4 someone complains about not being able to repro the test failure, so maybe add an opt out?

Reply:
Mostly because it's not an Inductor test. Not sure what refactor is appropriate here.